### PR TITLE
[TLI] Add missing ArmPL mappings

### PR DESCRIPTION
**Pull request by labrinea**
Adds TLI mappings for fixed and scalable vector variants of cospi(f), fmax(f), ilogb(f) and ldexp(f).

### CODE REVIEW

File: llvm/include/llvm/Analysis/VecFuncs.def [hunk 1]
```diff
@@ -943,6 +953,16 @@ TLI_DEFINE_VECFUNC("hypotf", "armpl_vhypotq_f32", FIXED(4), NOMASK, "_ZGV_LLVM_N
 TLI_DEFINE_VECFUNC("hypot", "armpl_svhypot_f64_x",  SCALABLE(2), MASKED, "_ZGVsMxvv")
 TLI_DEFINE_VECFUNC("hypotf", "armpl_svhypot_f32_x", SCALABLE(4), MASKED, "_ZGVsMxvv")
 
+TLI_DEFINE_VECFUNC("ilogb", "armpl_vilogbq_f64", FIXED(2), NOMASK, "_ZGV_LLVM_N2v")
+TLI_DEFINE_VECFUNC("ilogbf", "armpl_vilogbq_f32", FIXED(4), NOMASK, "_ZGV_LLVM_N4v")
+TLI_DEFINE_VECFUNC("ilogb", "armpl_svilogb_f64_x",  SCALABLE(2), MASKED, "_ZGVsMxv")
```
Comment (mgabka) [inline llvm/include/llvm/Analysis/VecFuncs.def:958]: I am not sure it is safe to add the mappings for the double version of ilogb, samel applies to ldexp, as the int element size does not match the double size, we will need to check it I think.

File: llvm/include/llvm/Analysis/VecFuncs.def [hunk 2]
```diff
@@ -943,6 +953,16 @@ TLI_DEFINE_VECFUNC("hypotf", "armpl_vhypotq_f32", FIXED(4), NOMASK, "_ZGV_LLVM_N
...
+TLI_DEFINE_VECFUNC("ilogbf", "armpl_svilogb_f32_x", SCALABLE(4), MASKED, "_ZGVsMxv")
+
+TLI_DEFINE_VECFUNC("ldexp", "armpl_vldexpq_f64", FIXED(2), NOMASK, "_ZGV_LLVM_N2vv")
```
Comment (paschalis-mpeis) [inline llvm/include/llvm/Analysis/VecFuncs.def:961]: Shouldn't the 2nd argument of ldexp(f) be a linear argument and not a vector?

Comment (mgabka) [reply llvm/include/llvm/Analysis/VecFuncs.def:961]: why do you think so?
the function is defined as: 
``
double  ldexp( double arg, int exp)
``
both arguments are passed by value, so from the vector function I am expecting to return a vector of results  where each element is multiplied by a different power of 2 (i.e different element of the input exp vector) .

Comment (labrinea) [reply llvm/include/llvm/Analysis/VecFuncs.def:961]: No, the amath header file suggests it is an integer vector.

Comment (paschalis-mpeis) [reply llvm/include/llvm/Analysis/VecFuncs.def:961]: Oh, right. For some reason I thought exp would be the same value for each element of arg.

@mgabka this explains your concern then, that we have 32bits float/int vectors and is OK, but what would actually happen with double/int vectors.

**Review by mgabka — APPROVED** [review]

**Pull request merged**